packages/material-ui/src/Collapse/Collapse.js

```diff
@@ -152,10 +156,12 @@ class Collapse extends React.Component {
       >
         {(state, childProps) => (
           <Component
+            aria-hidden={state === 'exited'}
```

Comment:
I vote for using `hidden` over `aria-hidden`. https://developer.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/. One of the implication of this attribute is to set `display: none`. This 1. Make the behavior consistent when `unmountOnExit: true` is used. 2. It removed the need for the custom existed style: simpler.

Reply:
I would prefer `hidden`. The current approach would be non-breaking. 

Using `hidden` would. While it would be consistent with `unmountOnExit: true` or `mountOnEnter: true` it's inconsistent with the current default values. It would break a lot of layouts (including our docs, ExpansionPanel etc).

I think we should come up with an API to control that behavior. The `hidden` heuristic is for Concurrent mode only so it's not that important at the moment. I would like that feature for v4 though in case this requires some breaking change.

Reply:
> It would break a lot of layouts (including our docs, ExpansionPanel etc).

I have tried the ExpansionPanel, it won't break it. The demo headers can be fixed by adding a `<div style={{ flexGrow: 1 }} />` element. Anything else? I have looked at the Bootstrap animations, they set display: none when the component is hidden.

Reply:
Even if this was true it is still a breaking change. You can't just "display: none;" and expect no breakage.

It's also not trivial to implement for all the transition components. Each handles thing slightly different and I'm not sure if this is actually intended or if those are all just workarounds that were not applied to the other components. Some use cloneElement, some 3! wrapper divs, some apply inline styles, others css classes. Some comments about SSR are either incorrect or omit important facts about the behavior they try to achieve.

Reply:
> Even if this was true it is still a breaking change.

It's targeting `next` though. (Or have you planned to back-port it to `master`?)

Reply:
Interesting, https://react-bootstrap.github.io/utilities/transitions/. Both bootstrap implementations use a different `in={false}` behaviors between the Fade and Collapse animation.
- Fade => visibility hidden
- Collapse => display none
I guess we could do that too. What do you think?

Reply:
> > Even if this was true it is still a breaking change.
> 
> It's targeting `next` though. (Or have you planned to back-port it to `master`?)

Just wanted to be on the same page that a fix here is a breaking change in other components. I would much rather focus on a a11y here and rethink proper `hidden` handling in a different PR. Changing `display` would add a lot of work.

Reply:
Ok